### PR TITLE
Add activitypub_post_object_type filter wrapping Post::get_type()

### PR DESCRIPTION
**Pull request by kraftbj**
## Context

Proposed as part of the FOSSE plugin's design plan for native Bluesky publishing — the design lives at [`Automattic/fosse#18`](https://github.com/Automattic/fosse/pull/18) (see `sdd/bluesky-native-publishing/`). The implementation here is a faithful expression of that plan.

- **If you disagree with WHAT this does** — the filter name, where it wraps, the single-return refactor, scope — the FOSSE SDD is the right place to push back. I'm happy to carry feedback in either direction.
- **Code-level concerns** (style, edge cases, test coverage, whether the refactor should be split out) are perfect for here.

Sibling PR in `Automattic/wordpress-atmosphere` introduces `atmosphere_is_short_form_post` so FOSSE can project a single user-level "always Note" preference onto both networks. Sibling atmosphere PR: https://github.com/Automattic/wordpress-atmosphere/pull/29

## Summary

Adds a new filter, `activitypub_post_object_type`, that wraps the return of [`Activitypub\Transformer\Post::get_type()`](https://github.com/Automattic/wordpress-activitypub/blob/trunk/includes/transformer/class-post.php#L453). Default behavior is unchanged for any consumer that does not hook the filter.

## Why a new filter (and not the existing `activitypub_transform_set_type`)

The existing `activitypub_transform_set_type` filter fires once per property set on the final activity object — it changes the wire-format `type` value going out, but it does not change what `Post::get_type()` returns to internal callers.

`Post::get_type()` is called in five places inside `Post::transform()` to drive downstream decisions:

- `set_former_type()` (tombstone handling, line 116)
- Content composition branching for Note vs Article (lines 538, 567)
- Content template selection (line 1062)
- Preview-generation guard (line 1140)

A downstream filter on `activitypub_transform_set_type` therefore leaves all five of those running with the original computed value, producing an object whose wire `type` says `Note` but whose content/template/preview were computed under `Article` — inconsistent state.

A filter on the discriminator itself is the clean fix: one override, applied uniformly to every caller.

## Changes

- `includes/transformer/class-post.php` — refactor `get_type()` to assign through all branches and return once at the bottom (trivially equivalent to the current early-return shape), then wrap the return value in `apply_filters( 'activitypub_post_object_type', $object_type, $this->item )`. Full PHPDoc on the new filter.
- `tests/phpunit/tests/includes/transformer/class-test-post.php` — new test `test_get_type_filter_overrides_computed_value()` verifying (1) the filter receives the computed default and the post, and (2) the filtered value replaces the computed value.
- `.github/changelog/add-post-object-type-filter` — minor/added entry.

## Test plan

- [x] `composer lint` — the two files I touched are clean (one pre-existing PHPCS error in `includes/cache/class-stats-image.php` is unrelated to this change and exists on trunk).
- [ ] PHPUnit via `wp-env` — I did not run locally (no wp-env set up in my environment). Confident in the change via PHP syntax check and by reading, but please let me know if CI surfaces anything.

Fixes DOTCOM-16839

### CODE REVIEW

File: includes/transformer/class-post.php [hunk 1]
```diff
@@ -454,25 +454,32 @@ protected function get_type() {
...
 
-		return $object_type;
+		/**
+		 * Filters the ActivityPub object type for a post.
+		 *
+		 * Allows downstream consumers to override the discriminator that
+		 * decides whether a post federates as Note, Article, or Page.
+		 * The filtered value propagates to all internal callers of
+		 * get_type() (content composition, attachment handling, content
+		 * templates, preview), not only the wire-format type property.
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-post.php]: Good catch. Reworded the docblock in c62ab159 to list the real internal callers (former_type/tombstone, summary/title, content template, preview guard) and drop the misleading attachment reference.

File: tests/phpunit/tests/includes/transformer/class-test-post.php [hunk 2]
```diff
@@ -233,6 +233,45 @@ public function test_get_type_respects_post_format_support() {
...
+			return 'Note';
+		};
+
+		\add_filter( 'activitypub_post_object_type', $callback, 10, 2 );
+
+		$transformer = new Post( $post );
+		$type        = $this->reflection_method->invoke( $transformer );
+
+		\remove_filter( 'activitypub_post_object_type', $callback, 10 );
+
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply tests/phpunit/tests/includes/transformer/class-test-post.php]: Agreed. Wrapped the invoke in try/finally in c62ab159 so the filter is removed even if the invocation throws.

**Review by jeherve — COMMENTED** [review]
Looks good to me, but probably just as a `patch` level fix.

**Review by pfefferle — APPROVED** [review]
Thanks!
